docstr_coverage/__init__.py

```diff
@@ -1,8 +1,23 @@
 ##################################################
 # Set __all__
 ##################################################
-from .coverage import get_docstring_coverage
+from .coverage import analyze, get_docstring_coverage
+from .ignore_config import IgnoreConfig
+from .printers import LegacyPrinter
+from .result_collection import (
+    AggregatedCount,
+    ExpectedDocstring,
+    FileCount,
+    ResultCollection,
+)
 
 __all__ = [
+    "analyze",
     "get_docstring_coverage",
+    "IgnoreConfig",
+    "LegacyPrinter",
+    "ResultCollection",
+    "AggregatedCount",
+    "ExpectedDocstring",
+    "FileCount",
```

Comment:
Do we really need to export all of these new classes used behind the scenes? I'm especially thinking about the classes from `result_collection.py`

Reply:
Well, the idea of the new  `analyze` function is to provide a nicer interface to pass in the ignore configuration (i.e., an `IgnoreConfig` instance) and to get back way more information than previously. This additional information comes as ResultCollection, which consists of `AggregatedCounts`, `ExpectedDocstring` and `FileCounts`. Thus for the user to be able to take advantage of this information in a nice way (for example, by explicitly using the types), I think exposing them makes sense.

Why would you prefer not to expose them? I'm sure I am missing something...

Reply:
That's a good point. Thank you for sharing your reasoning behind this. 
I suppose I'm thinking that although users will be interacting with these new classes when importing docstr-coverage as a module, they probably wouldn't actually ever be instantiating them. 
It seems to me that these classes are used almost exclusively by the docstr-coverage internals, and that a user wouldn't have a reason to create them directly. 
Please correct me if I'm missing something, which is entirely possible. Again, I appreciate your willingness to walk me through this

Reply:
I agree users do not have to create instances of said classes (except the `IgnoreConfig`). However, I am a big fan of type hints – and if our users want to use type hints when walking through the result collection, afaik, they have to import the classes. 
However, your question made me think more about what we should expose explicitly (in terms of expected use-cases). Let us take this discussion offline 😊

Reply:
Your logic is undeniable. Let's leave this as-is. Thank you for taking the time to discuss everything!